packages/reports/addon/components/report-actions/save-as.js

```diff
@@ -1,43 +1,42 @@
 /**
- * Copyright 2018, Yahoo Holdings Inc.
+ * Copyright 2020, Yahoo Holdings Inc.
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  *
  * Usage:
- *   {{#report-actions/save-as
- *      model=model
- *   }}
+ *   <ReportActions::SaveAs @model={{model}}>
  *      Inner template
- *   {{/report-actions/save-as}}
+ *   </ReportActions::SaveAs>
  */
 
 import Component from '@ember/component';
 import layout from '../../templates/components/report-actions/save-as';
-import { get, set, computed } from '@ember/object';
-
-export default Component.extend({
-  layout,
+import { action, set, computed } from '@ember/object';
+import { layout as templateLayout, tagName } from '@ember-decorators/component';
 
+@templateLayout(layout)
+@tagName('')
+export default class ReportActionSaveAs extends Component {
   /**
    * @property {Boolean} showModal - flag to control modal display
    */
-  showModal: false,
+  showModal = false;
 
   /**
    * @property {String} reportName - report name/title
    */
-  reportName: computed('model', function() {
-    return `(New Copy) ${get(this, 'model.title')}`.substring(0, 150);
-  }),
+  @computed('model')
```

Comment:
```suggestion
```
This is bug. `title` updates are not reflected. I would say forget about `computed` for now (it is not a big perf win)